itests/kit/deals.go

```diff
@@ -391,17 +390,14 @@ func (dh *DealHarness) ExtractFileFromCAR(ctx context.Context, file *os.File) st
 	ch, err := car.LoadCar(ctx, bserv.Blockstore(), file)
 	require.NoError(dh.t, err)
 
-	b, err := bserv.GetBlock(ctx, ch.Roots[0])
-	require.NoError(dh.t, err)
-
-	nd, err := ipld.Decode(b)
+	dserv := dag.NewDAGService(bserv)
+	nd, err := dserv.Get(ctx, ch.Roots[0])
 	require.NoError(dh.t, err)
```

Comment:
This works because we've moved the global decoders into the dagservice. If you actually wanted to go from a block to a go-ipld-format node though things are a bit annoying since you've got to go through go-ipld-legacy to register codecs.

It's not clear that this matters, but does this indicate we want either of:

Somewhere to easily reference the globals used by boxo/merkledag:
```
        d: = ipldlegacy.NewDecoder()
	d.RegisterCodec(cid.DagProtobuf, dagpb.Type.PBNode, ProtoNodeConverter)
	d.RegisterCodec(cid.Raw, basicnode.Prototype.Bytes, RawNodeConverter)
```

Or instead of deleting the registry code from go-ipld-format instead expose it as a non-global (e.g. build your own registry).

----

In the case here it happens to make more sense to load the node via the dagservice since you're passing it through to `NewUnixfsFile` which needs one anyway, however if you used the wrong dag-pb codec (e.g. go-codec-dapb wrapped in a `LegacyNode`) you'd end up with runtime breakages in `NewUnixfsFile`.